From e72203816f45859fbe2868fabb38b4056c6e895b Mon Sep 17 00:00:00 2001
From: Jun Lei <Jun.Lei@amd.com>
Date: Tue, 8 Dec 2015 10:30:43 -0500
Subject: [PATCH 0584/1110] drm/amd/dal: Fix-up HDMI 1080p light up on discrete
 ASIC

1.) Discrete does not have integrated table, yet DC depends on this to construct.  This dependency should be removed.

Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/core/dc_link.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
index ac39dff..e7a12e0 100644
--- a/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
+++ b/drivers/gpu/drm/amd/dal/dc/core/dc_link.c
@@ -769,10 +769,7 @@ static bool construct(
 		goto create_fail;
 	}
 
-	if (!dal_adapter_service_get_integrated_info(as, &info)) {
-		DC_ERROR("Failed to get integrated info!\n");
-		goto create_fail;
-	}
+	dal_adapter_service_get_integrated_info(as, &info);
 
 	for (i = 0; ; i++) {
 		if (!dal_adapter_service_get_device_tag(
-- 
2.7.4

